File: esports-backend/src/main/java/com/esports/dto/AuthResponse.java
package com.esports.dto;

import com.esports.entity.User;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AuthResponse {
    private Long id;
    private String name;
    private String email;
    private int role;
    private String token;
    private LocalDateTime createdAt;
    private boolean hasSelectedRole;

    public AuthResponse(Long id, String name, String email, User.Role role, String token, LocalDateTime createdAt) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.role = role.ordinal();
        this.token = token;
        this.createdAt = createdAt;
        this.hasSelectedRole = true;
    }

    public AuthResponse(Long id, String name, String email, User.Role role, String token, LocalDateTime createdAt, boolean hasSelectedRole) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.role = role.ordinal();
        this.token = token;
        this.createdAt = createdAt;
        this.hasSelectedRole = hasSelectedRole;
    }
}

File: esports-backend-dotnet/Upkart1/DTO/AuthResponse.cs
using Upkart1.Entities;
using Upkart1.Entities.Enums;

namespace Upkart1.DTO
{
    public class AuthResponse
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public int Role { get; set; }
        public string Token { get; set; }
        public DateTime CreatedAt { get; set; }
        public bool HasSelectedRole { get; set; }

        public AuthResponse() { }

        public AuthResponse(long id, string name, string email, Role role, string token, DateTime createdAt)
        {
            Id = id;
            Name = name;
            Email = email;
            Role = (int)role;
            Token = token;
            CreatedAt = createdAt;
            HasSelectedRole = true;
        }

        public AuthResponse(long id, string name, string email, Role role, string token, DateTime createdAt, bool hasSelectedRole)
        {
            Id = id;
            Name = name;
            Email = email;
            Role = (int)role;
            Token = token;
            CreatedAt = createdAt;
            HasSelectedRole = hasSelectedRole;
        }
    }
}
